### PR TITLE
Upgrade various dependencies

### PR DESCRIPTION
**Pull request by liblit**
* Upgrades that only changed one line in `gradle/libs.versions.toml`:
  * Upgrade Ant
  * Upgrade Apache Commons IO
  * Upgrade GSON
  * Upgrade Guava
  * Upgrade JUnit
  * Upgrade SLF4J
  * Upgrade Spotless
  * Upgrade ktfmt (Kotlin formatter)
* Upgrades that changed one line in `gradle/libs.versions.toml` plus trivial (whitespace-only) changes elsewhere:
  * [Update Google Java formatter](https://github.com/wala/WALA/commit/de8b3e4623d23d929351ab2acb03e796c11c59c8)
* Upgrades that changed one line in `gradle/libs.versions.toml` plus nontrivial changes elsewhere:
  * [Upgrade Error Prone](https://github.com/wala/WALA/commit/edc879a7166bd8f05aa5d71c93812adb973167dd)
  * [Upgrade gradle-download-task](https://github.com/wala/WALA/commit/0ed91b03e37caef17b0df324a2ec2ba341068cb4)

### CODE REVIEW

File: ide/jdt/test/src/test/java/com/ibm/wala/demandpa/driver/DemandCastChecker.java [hunk 1]
```diff
@@ -204,6 +204,7 @@ private static StateMachineFactory<IFlowLabel> makeStateMachineFactory() {
     return new ContextSensitiveStateMachine.Factory();
   }
 
+  @SuppressWarnings("ComparisonOutOfRange")
```
[the remark this answers is not in the document]
Comment (liblit) [reply ide/jdt/test/src/test/java/com/ibm/wala/demandpa/driver/DemandCastChecker.java:207]: > Did you look at the error message and conclude these were false positives?

I did. In each case, the code in question is comparing the current count or size of something to a `static final int` limit, where that limit is set to `Integer.MAX_VALUE`. It appears that the intent is to have no limit by default, but also to make it easy to impose a limit by changing how the `static final int` is initialized. Here's a typical example:

[Initialized field](https://github.com/wala/WALA/commit/edc879a7166bd8f05aa5d71c93812adb973167dd#diff-e80ddc2898d76253e8291c9b61bfa98fc4c085fae87b3547358475b4267ad18aR53):

```java
  private static final int MAX_SIZE = Integer.MAX_VALUE;
```

[Latent limit check](https://github.com/wala/WALA/commit/edc879a7166bd8f05aa5d71c93812adb973167dd#diff-e80ddc2898d76253e8291c9b61bfa98fc4c085fae87b3547358475b4267ad18aR224) that is always `true` unless the initialization of `MAX_SIZE` is changed:

```java
    if (size <= MAX_SIZE) {
```

I believe that these checks are correct when the limit is set to something below `Integer.MAX_VALUE`. I also believe that they _intentionally_ become vacuous in the default case when the limit is `Integer.MAX_VALUE`. So the warnings are correct in that each check is either always true or always false, but that is by design.

[the remark this answers is not in the document]
Comment (msridhar) [reply ide/jdt/test/src/test/java/com/ibm/wala/demandpa/driver/DemandCastChecker.java:207]: Thanks for the explanation!  I wish we didn't have to suppress this error-level check, but I'm not sure there is a better way to do things.  I guess we could set `MAX_SIZE` to `Integer.MAX_VALUE - 1`, which would make the check non-vacuous, or have a separate boolean variable to indicate whether a size limit is in place.  But both of those options seem like overkill for this issue.

For now, could you add a short comment in each spot explaining the suppression?

[the remark this answers is not in the document]
Comment (liblit) [reply ide/jdt/test/src/test/java/com/ibm/wala/demandpa/driver/DemandCastChecker.java]: 3e034cf8564d9f454182451cf79ad59a3fa22a51 improves our handling of these limit checks as follows:

* In `DemandCastChecker`, add the two `int`s in a way that avoids `int` overflow.
* In `ImmutableStack`, call a new helper function that centralizes the suppression and explains what's going on.

**Review by msridhar — APPROVED** [review]
Thanks for these updates!

**Pull request merged**